model_security_adjust_cft/security/product_security.xml

```diff
@@ -19,5 +19,35 @@
     <field name="domain_force"> [(1,'=',1)]</field>
 </record>
 
+<record id="product_own_lead_rule" model="ir.rule">
+    <field name="name" >Team Products (Sales Executive)</field>
+    <field name="model_id" ref="product.model_product_template"/>
+    <field eval="[(4,ref('base.group_sale_salesman'))]" name="groups"/>
+    <field name="domain_force"> [('categ_id.section_ids.member_ids','in',[user.id])]</field>
+    <field name="perm_unlink" eval="False"/>
+    <field name="perm_write" eval="False"/>
+    <field name="perm_create" eval="False"/>
+</record>
+
+<record id="product_team_lead_rule" model="ir.rule">
+    <field name="name" >Team Products (Sales Manager)</field>
+    <field name="model_id" ref="product.model_product_template"/>
+    <field eval="[(4,ref('group_sale_salesman_team_leads'))]" name="groups"/>
+    <field name="domain_force"> [('categ_id.section_ids.member_ids','in',[user.id])]</field>
+    <field name="perm_unlink" eval="False"/>
+    <field name="perm_write" eval="False"/>
+    <field name="perm_create" eval="False"/>
+</record>
```

Comment:
Is it necessary to have this record rule?  I guess it is not needed as I review the settings.  Please check.